### DISCUSSION TITLE
Cannot read properties of undefined (reading 'dark') in Jest

### DISCUSSION THREAD

**Discussion by eduardojustos** [quasarframework/quasar#12104] [General - Components / Directives / etc]
I am using Quasar with Vite Plugin + Jest and wrote the following test
```

describe('validate the form', () => {
    beforeEach(() => {
      jest.clearAllMocks();
    });

    const store = useStore();

    it('should call the validation functions', () => {
      const wrapper = mount(LicensePlate, { store });
      expect(wrapper.vm.canGoToNextStep()).toBe(false);
    });
  });
```

It tries to mount a page which uses QForm and QInput which I am injecting globally in jest (jest.init.js) by using

`config.global.components = {
  'q-form': QForm,
  'q-input': QInput,
};`

But I am receiving a "TypeError: Cannot read properties of undefined (reading 'dark')". Do I need to install any extensions or something like this?

**Comment by yusufkandemir** [comment]
Hi @eduardojustos, `dark` is meant to be a part of the `$q` object. The absence of `$q` means the `Quasar` plugin isn't installed on the testing app instance. It's normally done via `app.use(Quasar, { ... })` on the actual app, but it's a bit different with `@vue/test-utils`.

Follow https://github.com/quasarframework/quasar/issues/11424 for more details. Please don't forget to share your experience and ideas there! 👏

**Comment by eduardojustos** [comment]
I see. I ended up using the `installQuasarPlugin();` but still struggling to manage this issue.
It is saying now that I maybe need to add some things to the transformIgnorePatterns, in Jest's config.

`    Details:

    <myFolder>/node_modules/quasar/dist/quasar.esm.prod.js:6`
`    SyntaxError: Cannot use import statement outside a module`


But I have configured as the following: in jest.config.js

```
const esModules = ['quasar', 'quasar/lang', 'lodash-es', '@quasar/quasar-app-extension-testing-unit-jest'].join('|');

module.exports = {
  collectCoverageFrom: ['src/**/*.{vue,js,ts}', '!src/*.{js,ts}', '!src/typings/*'],
  setupFiles: [
    '<rootDir>/tests/jest.init.ts',
  ],

  moduleFileExtensions: [
    'vue',
    'js',
    'ts',
    'json',
    'jsx',
    'tsx',
  ],

  transform: {
    '^.+\\.ts$': 'ts-jest',
    '^.+\\.vue$': 'vue-jest',
  },

  transformIgnorePatterns: [`node_modules/(?!(${esModules}))`],

  moduleNameMapper: {
    '@/(.*)$': '<rootDir>/src/$1',
    '^quasar$': 'quasar/dist/quasar.esm.prod.js',
    'lodash-es': 'lodash',
  },

  globals: {
    'ts-jest': {
      tsconfig: '<rootDir>/tests/tsconfig.json',
    },
  },
};
```

So, quasar is being transformed, isn't it?
